Source/Core/Core/IOS/Network/Socket.cpp

```diff
@@ -331,11 +331,24 @@ void WiiSocket::Update(bool read, bool write, bool except)
       }
 
       // Fix blocking error codes
-      if (!nonBlock)
+      if (!nonBlock && it->net_type == IOCTL_SO_CONNECT)
       {
-        if (it->net_type == IOCTL_SO_CONNECT && ReturnValue == -SO_EISCONN)
+        switch (ReturnValue)
         {
+        case -SO_EAGAIN:
+        case -SO_EALREADY:
+        case -SO_EINPROGRESS:
+          if (DoTimeout())
```

Comment:
To me, DoTimeout makes it sound like you're performing the action of timing out something... Maybe this should just be split into two functions:

```cpp
void StartTimeoutIfNeeded()
{
  if (!timeout.has_value())
  {
    timeout = std::chrono::system_clock::now() +
              std::chrono::seconds(Config::Get(Config::MAIN_NETWORK_TIMEOUT));
  }
}

bool HasTimedOut() const { return timeout && std::chrono::system_clock::now() > *timeout; }
```